src/ucp/rndv/rndv.c

```diff
@@ -285,8 +285,7 @@ void ucp_rndv_req_send_ack(ucp_request_t *ack_req, ucp_request_t *req,
     ack_req->send.proto.status        = status;
     ack_req->send.proto.remote_req_id = remote_req_id;
     ack_req->send.proto.comp_cb       = ucp_request_put;
-    /* ack_req->send.state.uct_comp is used when ack operation is failed */
-    ucp_request_send_state_reset(ack_req, ucp_proto_comp_cb,
+    ucp_request_send_state_reset(ack_req, NULL,
```

Comment:
why removed? Looks like there will be a leak now when Ack is added to pending queue and then failed

Reply:
and moreover: it will try to call user completion CB

Reply:
> why removed? Looks like there will be a leak now when Ack is added to pending queue and then failed

it was removed, since Zcopy completion callback won't be called anymore.
the request will be release here:
https://github.com/openucx/ucx/blob/bf5c0904893a926e43282c3e2474912f33871b86/src/ucp/core/ucp_request.c#L431

> and moreover: it will try to call user completion CB

we have a check for NULL in fast-forwarding now:
https://github.com/openucx/ucx/blob/bf5c0904893a926e43282c3e2474912f33871b86/src/ucp/core/ucp_request.c#L418

Reply:
since this request was not created by the user, IMO better keep it as it was and release it by ucp_proto_comp_cb rather than by ucp_request_complete_send

Reply:
let's port https://github.com/yosefe/ucx/pull/55 ?

Reply:
maybe another PR, need to discuss.
in this PR, let's make minimal changes to fix comp_count>0 assert

Reply:
done